restapi/src/main/java/io/stargate/web/docsapi/dao/DocumentDB.java

```diff
@@ -120,6 +123,32 @@ public Keyspace getKeyspace(String keyspaceName) {
     return dataStore.schema().keyspaces();
   }
 
+  public Integer getTtlForDocument(String namespace, String collection, String docId) {
+    List<Row> result =
+        this.dataStore
+            .queryBuilder()
+            .select()
+            .function(
+                ImmutableList.of(
+                    QueryBuilderImpl.FunctionCall.ttl(DocsApiConstants.LEAF_COLUMN_NAME)))
+            .from(namespace, collection)
+            .where(BuiltCondition.of(DocsApiConstants.KEY_COLUMN_NAME, Predicate.EQ, docId))
+            .build()
+            .execute()
+            .join()
+            .rows();
+    try {
+      return result.stream()
+          .map(row -> row.getInt(String.format("ttl(%s)", DocsApiConstants.LEAF_COLUMN_NAME)))
+          .filter(Objects::nonNull)
+          .mapToInt(Integer::valueOf)
+          .max()
+          .getAsInt();
+    } catch (NoSuchElementException e) {
+      return null;
+    }
+  }
+
```

Comment:
We should avoid adding any new things in the `DocumentDB` and rather as such functionality to the `DocumentSearchService`.. I am also fine to have a separate `DocumentTTLService`, but point is that this is implemented in the same way as we do it for the document read:

* dedicated query builder
* tests for query builder
* reactive service implementation that does separate prepare and execute, and returns a Single<Integer> (can be empty in case there is no ttl)
* test for that service implementation